src/wlserver.hpp

```diff
@@ -150,9 +150,9 @@ struct wlserver_t {
 	bool mouse_constraint_requires_warp = false;
 	pixman_region32_t confine;
 	struct wlr_pointer_constraint_v1 *mouse_constraint = nullptr;
-	uint64_t ulLastMovedCursorTime = 0;
-	bool bCursorHidden = true;
-	bool bCursorHasImage = true;
+	std::atomic<uint64_t> ulLastMovedCursorTime = 0;
+	std::atomic<char> bCursorHidden = true;
```

Comment:
Why char?

Reply:
> Why char?

For some reason, you can't do `fetch_or` on a boolean atomic...
Which in the one case where I use it, I can't cleanly replace it with something like an atomic exchange
only integral atomics can do stuff like `fetch_or`
https://en.cppreference.com/w/cpp/atomic/atomic#Specializations_for_integral_types

So I use char, since a bool, when stored in a struct/class (in most implementations), actually occupies the same amount of memory as a char.
Would you prefer I use `uint8_t` instead of char?

I believe most modern cpus should support lock-free atomics on 8 bit values, so I think it should be fine?